File: ServerSubscriptionManager/Models/Payment.cs
namespace ServerSubscriptionManager.Models
{
    public class Payment
    {
        public long Id { get; set; }
        public required decimal Amount { get; set; }
        public DateOnly Date { get; set; }
        public required long PaymentMethodId { get; set; }
        public PaymentMethod? PaymentMethod { get; set; }
        public bool Valid { get; set; } = false;
        public required long UserId { get; set; }
        public User? User { get; set; }
    }
}

File: web/src/app/models/payment.ts
import { PaymentMethod } from "./payment-method";
import { User } from "./user";

export interface Payment {
    id: number;
    amount: number;
    date: string;
    paymentMethodId: number;
    paymentMethod: PaymentMethod;
    valid: boolean;
    userId: number;
    user?: User;
}